File: 1528-shuffle-string/1528-shuffle-string.js
/**
 * @param {string} s
 * @param {number[]} indices
 * @return {string}
 */
var restoreString = function(s, indices) {
    // console.log(s);
    // console.log(indices);
    let reshuffleArr = [];
    for(i=0; i<indices.length; i++){
        reshuffleArr[indices[i]] = s[i];
        // console.log(s[i]);
        // console.log(reshuffleArr)
    }
    // console.log(reshuffleArr);
    return reshuffleArr.join("");
    
};

File: 1528-shuffle-string/1528-shuffle-string.java
class Solution {
    public String restoreString(String s, int[] indices) {
        char[] charArr = new char[indices.length];
        for(int i=0; i<indices.length; i++){
            charArr[indices[i]] = s.charAt(i);
        }
        String ans = String.valueOf(charArr);
        return ans;
    }
}